pandas/core/generic.py

```diff
@@ -721,12 +721,37 @@ def transpose(self, *args, **kwargs):
         new_axes = self._construct_axes_dict_from(
             self, [self._get_axis(x) for x in axes_names]
         )
-        new_values = self.values.transpose(axes_numbers)
-        if kwargs.pop("copy", None) or (len(args) and args[-1]):
-            new_values = new_values.copy()
+
+        if (
+            self._is_homogeneous_type
+            and len(self._data.blocks)
+            and is_extension_array_dtype(self._data.blocks[0].dtype)
+        ):
+            kwargs.pop("copy", None)  # by definition, we're copying
+            dtype = self._data.blocks[0].dtype
+            arr_type = dtype.construct_array_type()
+
+            # Slow, but unavoidable with 1D EAs.
+            new_values = []
+            for i in range(len(self)):
```

Comment:
I'm rethinking this approach. This results in `n_rows * n_columns` `__getitem__`s. My intent was to avoid going through a 2D object-dtype ndarray. But we're essentially doing that with lists. So I think it'll be better to just do `.values.T` and then rebuild the EAs from the object-dtype array.